Update tx build with voting and actions

Jimbo4350 opened the pull request:
# Description

Resolves #42 and #36
# Changelog

```yaml
- description: |
    Update cardano-cli with the ability to create votes and governance actions.
    Update tx build with the ability to specify votes and governance actions.
  compatibility: compatible
  type: feature
```

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

Jimbo4350 committed:
Implement Cardano.CLI.Conway.Parsers
Move more parsers in to Cardano.CLI.Common.Parsers

Jimbo4350 committed:
Implement ConwayProposal type
Move more parsers to Cardano.CLI.ConwayParsers

Jimbo4350 committed:
Wired up create-vote and create-constitution commands

Jimbo4350 committed:
Implement
- Cardano.CLI.Conway.Commands/Types
- ConwayVote
Propagate new TxBodyContent votes and proposals records

newhoggy committed:
Parameterise Certificate with era

carbolymer commented on cardano-cli/src/Cardano/CLI/Shelley/Commands.hs:
```diff
@@ -48,20 +49,19 @@ module Cardano.CLI.Shelley.Commands
-import           Data.Time.Clock
-
+import           Cardano.Chain.Common (BlockCount)
+import           Cardano.CLI.Conway.Parsers
```
Is there a way to avoid importing newer eras from modules for older eras? This makes reasoning what can be imported where quite hard.

Jimbo4350 committed:
Wiring up votes and new constitutions to build command
Need to expose more types from cardano-api but certificates era
parameterization was merged so we need to merge that first in the cli so
I can use a cardano-api SRP

Jimbo4350 committed:
Bump cardano-api to accomodate for coercing a StakePoolKey to a StakeKey

Jimbo4350 committed:
Update readCddl Tx to handle Conway era txs

Jimbo4350 committed:
Bump cardano-api and propagate updated types

disassembler committed:
Fix the name of crypt32 library in crypton-x509-system

disassembler committed:
cardano-api: remove srp

disassembler committed:
hackageNix: bump

newhoggy committed:
Style improvements

newhoggy committed:
Fix tests

newhoggy committed:
Prefer asum over <|>, prefer mconcat over <>

newhoggy committed:
Update CABAL_CACHE_VERSION
